tests/virtualization/universal/hotplugging.pm

```diff
@@ -136,14 +136,16 @@ sub test_add_vcpu {
     }
     return if (is_xen_host && $guest =~ m/hvm/i);    # not supported on HVM guest
 
+    if ($sles_running_version eq '15' && $sles_running_sp eq '4' && is_xen_host && is_fv_guest($guest)) {
+        record_info 'Skip vCPU hotplugging', 'bsc#1188898 vCPU hotplugging does no work on SLE FV guest on SLE15-SP4 host';
+        return;
+    }
+
     # Ensure guest CPU count is 2
     die "Setting vcpus failed" unless (set_vcpus($guest, 2));
     assert_script_run("ssh root\@$guest nproc | grep 2", 60);
     # Add 1 CPU
-    if ($sles_running_version eq '15' && $sles_running_sp eq '4' && is_xen_host && is_fv_guest($guest)) {
-        record_soft_failure('bsc#1188898 Failed to set live vcpu count on fv guest on 15-SP4 Xen host');
-    }
-    elsif ($sles_running_version eq '15' && $sles_running_sp eq '3' && is_xen_host && is_fv_guest($guest)) {
+    if ($sles_running_version eq '15' && $sles_running_sp eq '3' && is_xen_host && is_fv_guest($guest)) {
         record_soft_failure('bsc#1180350 Failed to set live vcpu count on fv guest on 15-SP3 Xen host');
     }
     else {
```

Comment:
What about move
```
# Ensure guest CPU count is 2
die "Setting vcpus failed" unless (set_vcpus($guest, 2));
assert_script_run("ssh root\@$guest nproc | grep 2", 60);
```
into the "else" branch ? Then you do not need to change anything else.

Reply:
Refer to the test_add_vcpu {} roadmap, there should be **# Ensure guest CPU count is 2** firstly and then **# Add 1 CPU**.

Reply:
For the test, Yes. But if the test does not need to run, then "Ensure guest CPU count is 2" is not needed either.

Reply:
yes, we need to figure out the easy and simple way to skip this `test_add_vcpu {}` on sles xen host with FV guest.  I totally agree on your POV here. But, there were more `if` and `else` branch under `test_add_vcpu {}`.

Reply:
I see. But I think, at least, we can make the if...elsif...else better by making the change I suggested above. Maybe @pdostal can provide relevant input here about how this looks like from their testing POV. If he does not object either, we can make the change ?

Reply:
Awesome! yes, I totally agree on your suggestion as my mentioned. also, we'd better to figure out the easy and simple way to maintain this test module with QAM together. 
btw, regarding the bsc#1188898 issue on sles15sp4 was a little bit more special as my mentioned. Due to compare with the latest sles15sp4 build 31.2 and build 26.1,  `bsc#1188898` had a little changed for `bsc#1180350`. if still used with `if... elsif ...else` means that move so much of codes to keep the same  logical structure for this test module POV.

Reply:
I  understand your concern. You want to keep the original test structure. Right ?

Reply:
due to the current bsc#1188898 issue on sles15sp4, plan to keep the original test structure for this hotplugging test module. 
For now, I am not clear if there was still new changes for **bsc#1188898** on sles15sp4 again during `hotplugging::test_add_vcpu {}`. 
Anyway, hope our developer fix this **bsc#1188898** on sles15sp4 quickly.

Reply:
I do not think this involves any specific bug or bug fix. Basically the "else" branch should include everything needs to be set and test. If any test scenario should be skipped then it should be in another "if" or "elsif" branch to skip it. Currently the "Ensure guest CPU count is 2" part wanders around instead of  being the pre-conditions to sit together with the actual test part. In other words, there is no point to "ensure anything" if "actual test part should be skipped".

Reply:
yes, I see.  As the purpose for this PR to skip `test_add_vcpu {}` for **bsc#1188898** on sles15sp4. 
Also, Due to currently the "Ensure guest CPU count is 2" part wanders around instead of being the pre-conditions to sit together with the actual test part. maybe we can continue this season on ticket [Separate hotplugging.md into _nic, _cpu, _mem files](https://progress.opensuse.org/issues/96230).